messaging/src/main/java/ids/messaging/endpoint/MessageController.java

```diff
@@ -126,13 +140,23 @@ public ResponseEntity<MultiValueMap<String, Object>> handleIDSMessage(
                                              "Header was missing!"));
             }
 
+            final var headerBytes = IOUtils.toByteArray(headerPart.getInputStream());
+            if (Boolean.TRUE.equals(logIncoming)) {
```

Comment:
Why not shifting this processing a little bit more below where the header is already present as ByteArrayInputStream (line157).

Reply:
Thats the same reason as before, headerPart.getInputStream() consumes the stream, so it needs to be rebuild as Stream out of its bytes in line 157. If you now move it below 157 and remove this kind of copy-logic, you would have an empty header-stream if logging the message is enabled.